lib/Varien/Data/Form/Element/Abstract.php

```diff
@@ -214,7 +214,7 @@ public function removeClass($class)
      */
     protected function _escape($string)
     {
-        return htmlspecialchars($string, ENT_COMPAT);
+        return is_string($string) ? htmlspecialchars($string, ENT_COMPAT) : $string;
```

Comment:
In general I'm happy with your PR mainly relying on added typechecks to avoid problems with PHP 8.1. But in this case I would like to argue that an explicit cast to string `htmlspecialchars((string)$string, ENT_COMPAT)` would be more appropriate to use (because we don't want to introduce parameter types here to avoid breaking things). 
The method clearly escapes a string, the parameter name and the phpdoc type point in that direction. While a `is_string` is not taxing the check will be performed a lot of times judging by its location in Varien Lib.